pkg/ast/pipesearch/wsSearchHandler.go

```diff
@@ -333,6 +347,7 @@ func processCompleteUpdate(conn *websocket.Conn, sizeLimit, qid uint64, aggs *st
 			log.Errorf("qid=%d, processCompleteUpdate: failed to write error response to websocket! err: %+v", qid, wErr)
 		}
 	}
+	LogGlobalErrors(qid)
 }
```

Comment:
I'm worried about cases that don't get here, and so we don't ever print the error messages (e.g., the query times out or is canceled). Maybe a better way to do this would be to `defer` it in `ProcessPipeSearchWebsocket()` once we have the `qid`. On that note, you'll also want to ensure this gets called when we handle queries with `ProcessPipeSearchRequest()` (that's the handler for non-UI queries, like if you run a query via curl)

Reply:
I have moved the logging in DeleteQuery and CancelQuery. We cannot directly defer this function as query info would be deleted after DeleteQuery. This should handle the ProcessPipeSearchRequest case also because after Executing the query it calls DeleteQuery.